doc/tutorials/imgproc/imgtrans/distance_transformation/distance_transform.markdown

```diff
@@ -0,0 +1,165 @@
+Image Segmentation with Distance Transform and Watershed Algorithm {#tutorial_distance_transform}
+=============
+
+Goal
+----
+
+In this tutorial you will learn how to:
+
+-   Use the OpenCV function @ref cv::filter2D in order to perform some laplacian filtering for image sharpening
+-   Use the OpenCV function @ref cv::distanceTransform in order to obtain the derived representation of a binary image, where the value of each pixel is replaced by its distance to the nearest background pixel
+-   Use the OpenCV function @ref cv::watershed in order to isolate objects in the image from the background
+
+Theory
+------
+
+Code
+----
+
+This tutorial code's is shown lines below. You can also download it from
+    [here](https://github.com/Itseez/opencv/tree/master/samples/cpp/tutorial_code/ImgTrans/imageSegmentation.cpp).
+@includelineno samples/cpp/tutorial_code/ImgTrans/imageSegmentation.cpp
+
+Explanation / Result
+--------------------
+
+-#  Load the source image and check if it is loaded without any problem, then show it:
+    @code{.cpp}
+
+    Mat src = imread(argv[1]);
+
+    if (!src.data)
+        return -1;
+
+    imshow("Source Image", src);
+    @endcode
+![](images/source.jpeg)
+
+-#  Then if we have an image with white background, it is good to tranform it black. This will help us to desciminate the foreground objects easier when we will apply the Distance Transform:
+    @code{.cpp}
+
+    for( int x = 0; x < src.rows; x++ ) {
+      for( int y = 0; y < src.cols; y++ ) {
+          if ( src.at<Vec3b>(x, y) == Vec3b(255,255,255) ) {
+            src.at<Vec3b>(x, y)[0] = 0;
+            src.at<Vec3b>(x, y)[1] = 0;
+            src.at<Vec3b>(x, y)[2] = 0;
+          }
+        }
+    }
+
+    imshow("Black Background Image", src);
+    @endcode
+![](images/black_bg.jpeg)
+
+-#  Afterwards we will sharp our image in order to acute the edges of the foreground objects. We will apply a laplacian filter with a quite strong filter (an approximation of second derivative):
+    @code{.cpp}
+
+    Mat kernel = (Mat_<float>(3,3) <<
+            1,  1, 1,
+            1, -8, 1,
+            1,  1, 1);
+
+    Mat imgLaplacian;
+    Mat sharp = src;
+    filter2D(sharp, imgLaplacian, CV_32F, kernel);
+    src.convertTo(sharp, CV_32F);
+    Mat imgResult = sharp - imgLaplacian;
+
+    imgResult.convertTo(imgResult, CV_8UC3);
+    imgLaplacian.convertTo(imgLaplacian, CV_8UC3);
+
+    imshow( "Laplace Filtered Image", imgLaplacian );
+    imshow( "New Sharped Image", imgResult );
+    @endcode
+![](images/laplace.jpeg)
+
+![](images/sharp.jpeg)
```

Comment:
List numeration is restarted after these images, to avoid it align all image links with 4 spaces to put them in the same list item.